packages/markdown-support/src/index.ts

```diff
@@ -11,52 +11,43 @@ import markdownToHtml from 'remark-rehype';
 // import smartypants from '@silvenon/remark-smartypants';
 import rehypeStringify from 'rehype-stringify';
 
-export interface MarkdownRenderingOptions extends Partial<AstroMarkdownOptions> {
-  $?: {
-    scopedClassName: string | null;
-  };
-  mode: 'md' | 'astro-md';
-}
+export { AstroMarkdownOptions, MarkdownRenderingOptions };
 
 /** Internal utility for rendering a full markdown file and extracting Frontmatter data */
 export async function renderMarkdownWithFrontmatter(contents: string, opts?: MarkdownRenderingOptions | null) {
   // Dynamic import to ensure that "gray-matter" isn't built by Snowpack
   const { default: matter } = await import('gray-matter');
   const { data: frontmatter, content } = matter(contents);
-  const value = await renderMarkdown(content, { ...opts, mode: 'md' });
+  const value = await renderMarkdown(content, opts);
   return { ...value, frontmatter };
 }
 
 /** Shared utility for rendering markdown */
 export async function renderMarkdown(content: string, opts?: MarkdownRenderingOptions | null) {
-  const { $: { scopedClassName = null } = {}, mode = 'astro-md', footnotes: useFootnotes = true, gfm: useGfm = true } = opts ?? {};
+  const { $: { scopedClassName = null } = {}, remarkPlugins = [], rehypePlugins = [] } = opts ?? {};
   const { headers, rehypeCollectHeaders } = createCollectHeaders();
 
-  let parser = unified().use(markdown).use(remarkCodeBlock());
+  let parser = unified().use(markdown);
 
   if (scopedClassName) {
     parser = parser.use(scopedStyles(scopedClassName));
   }
+  const loadedRemarkPlugins = await Promise.all(remarkPlugins);
```

Comment:
If we allow the config to provide string specifiers this could be: 

```js
const loadedRemarkPlugins = await Promise.all(remarkPlugins.map(plugin => typeof plugin === 'string' ? import(plugin) : plugin));
```